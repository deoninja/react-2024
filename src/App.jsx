import { useEffect, useState } from 'react';
import { UserContext } from './utils/contexts/UserContext';
import { PostContainer } from './components/PostContainer';
import { PostContentButtons } from './components/PostContentButtons';
import { useFetchUser } from './utils/hooks/useFetchUser';
import { Outlet, Link, useNavigate } from 'react-router-dom';

export default function App() {
  const { user, loading, error } = useFetchUser(2);

  // console.log(user, loading, error);

  const [userData, setUserData] = useState();

  const navigate = useNavigate();

  useEffect(() => {
    if (!loading && !error && user) setUserData(user);
    // navigate('/users');
  }, [loading, error, user, navigate]);

  return (
    <>
      <nav>
        <ul>
          <li>
            <Link to='/'>Home</Link>
          </li>
          <li>
            <Link to='/users'>Users</Link>
          </li>
          <li>
            <Link to='/blog-post'>Blog</Link>
          </li>
        </ul>
      </nav>

      <div>
        <label htmlFor='data'>Enter data</label>
        <input
          type='text'
          id='data'
          onChange={(e) => {
            if (e.target.value.length > 10) {
              navigate('/blog-posts');
            }
          }}
        />
      </div>

      <UserContext.Provider value={{ ...userData, setUserData }}>
        <div>{loading ? 'Loading...' : <PostContainer />}</div>
      </UserContext.Provider>
      <Outlet />
    </>
  );
}
