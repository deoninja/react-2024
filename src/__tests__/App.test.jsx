import { expect, it, describe } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from '../App';

describe('App', () => {
  it('should render users', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
  describe('Edit Button is Clicked', () => {
    it('should render save button when edit is clicked', async () => {
      render(<App />);
      const editButton = screen.getByRole('button', { name: 'Edit' });
      await userEvent.click(editButton);
      const saveButton = screen.getByRole('button', { name: 'Save' });
      expect(saveButton).toBeInTheDocument();
    });
  });

  it('should render save button when edit is clicked', async () => {
    render(<App />);
    const editButton = screen.getByRole('button', { name: 'Edit' });
    await userEvent.click(editButton);
    expect(
      screen.getByRole('textbox', { name: 'username' })
    ).toBeInTheDocument();
    expect(screen.getByRole('textbox', { name: 'email' })).toBeInTheDocument();
  });
});
