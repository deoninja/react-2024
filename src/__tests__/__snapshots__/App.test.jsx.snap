// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should render users 1`] = `
<div>
  <div>
    <div>
      <button>
        Edit
      </button>
      <button>
        Delete
      </button>
    </div>
    <div>
      <b>
        ID:
      </b>
      <span>
        1
      </span>
      <br />
      <b>
        Username:
      </b>
      <span>
        johndoe
      </span>
      <br />
      <b>
        Email:
      </b>
      <span>
        l3X2k@example.com
      </span>
    </div>
  </div>
  <div>
    <div>
      <button>
        Edit
      </button>
      <button>
        Delete
      </button>
    </div>
    <div>
      <b>
        ID:
      </b>
      <span>
        2
      </span>
      <br />
      <b>
        Username:
      </b>
      <span>
        janedoe
      </span>
      <br />
      <b>
        Email:
      </b>
      <span>
        j8wKu@example.com
      </span>
    </div>
  </div>
</div>
`;
